javascript-carousel/main.js

```diff
@@ -0,0 +1,133 @@
+var imageList = document.querySelectorAll('img');
+var buttonsList = document.querySelectorAll('.button-image');
+var timerID = null;
+
+// image change every so and so seconds
+function imageChangeWithInterval() {
+  imageChangeForward();
+  timerID = setTimeout(imageChangeWithInterval, 1300);
+}
+// imageChangeWithInterval();
+
+var activeIndex = 0;
+
+function displayImage(index) {
+  for (var i = 0; i < imageList.length; i++) {
+    if (i !== index) {
+      imageList[i].classList.add('hidden');
+      buttonsList[i].classList.remove('current-button');
+    }
+    imageList[index].classList.remove('hidden');
+    buttonsList[index].classList.add('current-button');
+  }
+  activeIndex = index;
+}
+
+function getNextIndex() {
+  if (activeIndex === imageList.length - 1) {
+    return 0;
+  }
+  return activeIndex + 1;
+}
+
+function getPreviousIndex() {
+  if (activeIndex === 0) {
+    return imageList.length - 1;
+  }
+  return activeIndex - 1;
+}
+
+// image changes going forward
+function imageChangeForward() {
+  var currentImage = document.querySelector('.current-pokemon');
+  var currentButton = document.querySelector('.current-button');
+
+  var nextImage = currentImage.nextElementSibling;
+  var nextButton = currentButton.nextElementSibling;
+
+  if (currentImage.id === 'jigglypuff' || currentButton.id === 'lastbutton') {
+    nextImage = document.querySelector('#bulbasaur');
+    nextButton = document.querySelector('#firstbutton');
+  }
+
+  currentImage.classList.remove('current-pokemon');
+  currentImage.classList.add('hidden');
+  nextImage.classList.remove('hidden');
+  nextImage.classList.add('current-pokemon');
+
+  currentButton.classList.remove('current-button');
+  nextButton.classList.add('current-button');
+}
+
+// click right arrow moves image reel forward
+function clickRightChevron(event) {
+  if (!event.target.matches('.fa-chevron-right')) {
+    return;
+  }
+  imageChangeForward();
+
+  clearTimeout(timerID);
+  timerID = setTimeout(imageChangeWithInterval, 1300);
+}
+document.addEventListener('click', clickRightChevron);
+
+// click left arrow moves image backward
+function clickLeftChevron(event) {
+  if (!event.target.matches('.fa-chevron-left')) {
+    return;
+  }
+
+  var currentImage = document.querySelector('.current-pokemon');
+  var currentButton = document.querySelector('.current-button');
+
+  var previousImage = currentImage.previousElementSibling;
+  var previousButton = currentButton.previousElementSibling;
+
+  if (currentImage.id === 'bulbasaur' || currentButton.id === 'firstbutton') {
+    previousImage = document.querySelector('#jigglypuff');
+    previousButton = document.querySelector('#lastbutton');
+  }
+
+  currentImage.classList.remove('current-pokemon');
+  currentImage.classList.add('hidden');
+  previousImage.classList.remove('hidden');
+  previousImage.classList.add('current-pokemon');
+
+  currentButton.classList.remove('current-button');
+  previousButton.classList.add('current-button');
```

Comment:
You can use your functions here too, to get the previous index and display the correct image.